File: 프로그래머스/lv1/12921. 소수 찾기/소수 찾기.py
def solution(n):
    answer = 0
    
    result = [False for _ in range(n + 1)]
    
    for num in range(2, n + 1):
        if not result[num]:
            answer += 1
            
            count = 1
            while num * count <= n:
                result[num * count] = True
                count += 1
                
    return answer

File: 프로그래머스/2/42839. 소수 찾기/소수 찾기.js
const isPrime = (num) => {
    if (num <= 1) return false;
    
    for (let i = 2; i <= parseInt(num**(1/2)); i++) {
        if (num % i === 0) return false;
    }
    
    return true;
}

function solution(numbers) {
    const visited = Array(numbers.length).fill(false);
    const nums = new Set();
    
    const dfs = (num) => {
        if (num) nums.add(Number(num));
        
        for (let i = 0; i < numbers.length; i++) {
            if (!visited[i]) {
                visited[i] = true;
                dfs(num + numbers[i]);
                visited[i] = false;
            }
        }
    };
    
    dfs("");
    
    const answer = [...nums].filter((num) => isPrime(num)).length;
    
    return answer;
}